File: docs/src/pages/GuidePage/content/calculation/code/calculate_project.py
from lcax import calculate_project, CalculationOptions, LifeCycleModule, ImpactCategoryKey, Project
from pathlib import Path

project = Project.loads(Path("project.json").read_text())

options = CalculationOptions(
    reference_study_period=50,
    life_cycle_modules=[LifeCycleModule.A1A3, LifeCycleModule.A4, LifeCycleModule.B4, LifeCycleModule.C3, LifeCycleModule.C4],
    impact_categories=[ImpactCategoryKey.Gwp],
    overwrite_existing_results=True
)

# The results are added to the project object
project = calculate_project(project, options)

File: docs/src/pages/GuidePage/content/calculation/code/calculateProject.ts
import { calculateProject, CalculationOptions, LifeCycleModule, ImpactCategoryKey } from 'lcax'
import fs from 'fs'

const project_data = fs.readFileSync('project.json')
const project = JSON.parse(project_data)

const options: CalculationOptions = {
  referenceStudyPeriod: 50,
  lifeCycleModules: ['a1a3', 'a4', 'b4', 'c3', 'c4'] as LifeCycleModule[],
  impactCategories: ['gwp'] as ImpactCategoryKey[],
  overwriteExistingResults: true,
}

const calculatedProject = calculateProject(project, options)
